back-end/src/main/java/index/SSVInMemoryIndexer.java

```diff
@@ -205,6 +208,55 @@ private void setCommonVariables() throws Exception {
         Main.getProject().addBGRP(rpKey, "roughN", String.valueOf(rawRows.size()));
     }
 
+    private void getGeoCoords() throws Exception {
+        DbConnector.closeConnection(ssv.getDb());
+        rawDbStmt = DbConnector.getStmtByDbName(ssv.getDb());
+        String sql =
+                "ALTER TABLE " + ssv.getRawTable() + " ADD COLUMN IF NOT EXISTS kyrix_geo_x float;";
+        System.out.println(sql);
+        rawDbStmt.executeUpdate(sql);
+
+        sql = "ALTER TABLE " + ssv.getRawTable() + " ADD COLUMN IF NOT EXISTS kyrix_geo_y float;";
+        System.out.println(sql);
+        rawDbStmt.executeUpdate(sql);
+
+        sql = "DROP FUNCTION get_coord";
+        System.out.println(sql);
+        rawDbStmt.executeUpdate(sql);
+
+        sql =
+                "CREATE OR REPLACE FUNCTION get_coord(lat float, lon float) returns jsonb as $$"
+                        + ssv.getGetCoordinatesFromLatLonBody()
+                                .replaceAll(
+                                        "REPLACE_ME_geo_initial_level",
+                                        String.valueOf(ssv.getGeoInitialLevel()))
+                                .replaceAll(
+                                        "REPLACE_ME_geo_initial_center_lat",
+                                        String.valueOf(ssv.getGeoLat()))
+                                .replaceAll(
+                                        "REPLACE_ME_geo_initial_center_lon",
+                                        String.valueOf(ssv.getGeoLon()))
+                                .replaceAll(
+                                        "REPLACE_ME_top_level_width",
+                                        String.valueOf(ssv.getTopLevelWidth()))
+                                .replaceAll(
+                                        "REPLACE_ME_top_level_height",
+                                        String.valueOf(ssv.getTopLevelHeight()))
+                        + " $$ language plv8;";
+        System.out.println(sql);
+        rawDbStmt.executeUpdate(sql);
+
+        sql =
+                "UPDATE "
+                        + ssv.getRawTable()
```

Comment:
SQL injection... 

suggestion: instead of using injection safe parameters (ugly pain), rename getRawTable() to add something indicating safety e.g. getRawTableSqlSafe(), and then inside the function, restrict the character set to ensure no injection, I.e. you don't care about during weird table names. Wrapping also works (e.g. sqlSafe(getRawTable()) though it requires told support to ensure that you don't forget to do this everywhere...

Reply:
Makes sense. Since `getRawTable` returns a table name we get from compiler - the validation should be done in compiler. 

I think this falls into the bigger issue #15, where we need to do better validation of user spec to deal with "overly clever" developers.